### ISSUE TITLE
Failure to analyze when the symbol table is disabled on darwin/arm64/go (1.22,1.23)/cgo

### ISSUE THREAD

**Issue by nikpivkin** [Zxilly/go-size-analyzer#52]
Hi, thanks for this tool!

Steps to reproduce:
- git clone https://github.com/aquasecurity/trivy.git && cd trivy
- go build -ldflags "-s" -o trivy ./cmd/trivy/main.go
- gsa ./trivy --verbose
```sh
level=INFO msg="Parsing binary..."
level=INFO msg="Parsing binary done"
level=INFO msg="Finding build info..."
level=INFO msg="Build info found"
level=INFO msg="Loading sections..."
level=INFO msg="Loading sections done"
level=INFO msg="Loading packages..."
level=ERROR msg="Fatal error: failed to find runtime.text symbol: moduledata structure not found"
```

But if I add an environment variable `CGO_ENABLED=0` to disable cgo, the analysis runs successfully.

OS: macos m1
Go: 1.22

**Comment by Zxilly** [comment]
Do you compile with the target darwin-arm64? If so this is an known issue.
https://github.com/Zxilly/go-size-analyzer/blob/master/scripts/skip.txt

**Comment by Zxilly** [comment]
A temporary workaround is to keep the symbol table, and the dwarf can be removed as it didn't be used right now.

**Comment by nikpivkin** [comment]
Yes, the target is darwin-arm64. Why does this work with CGO_ENABLED=0?

**Comment by nikpivkin** [comment]
I got it, thanks.

**Comment by nikpivkin** [comment]
@Zxilly I can provide dumps if it can help. I've been trying to research this, but I haven't gotten too far.

**Comment by Zxilly** [comment]
It's an error triggered by an upstream library, and I'm sure it's fixable. And I already have some samples I can use to reproduce the error.
But I've been busy looking for a job lately. Maybe I'll have time to look at this in a few months.
You can follow https://github.com/goretk/gore for progress.

**Comment by Zxilly** [comment]
related: https://github.com/goretk/gore/issues/96

**Comment by Zxilly** [comment]
Fixed in v1.7.3
